### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_redux-list-of-todos/)

### CODE REVIEW

**Review by v-vytrykush — COMMENTED** [review]
Good job!

File: src/components/CurrentUser/CurrentUser.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { getUserSelector, setUserAction, setSelectedTodoId } from '../../store';
+import './CurrentUser.scss';
+
+export const CurrentUser: React.FC = () => {
+  const dispatch = useDispatch();
+  const user = useSelector(getUserSelector);
+
+  return (
+    <div className="CurrentUser">
+      {user === null ? ('No user selected') : (
```
Comment (Denchepornyuk) [inline src/components/CurrentUser/CurrentUser.tsx:13]: move this check to separate `if` above return

File: src/components/CurrentUser/CurrentUser.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        <>
+          {user.id === 0 ? (
```
Comment (Denchepornyuk) [inline src/components/CurrentUser/CurrentUser.tsx:15]: I believe you can remove this check

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  setSelectedTodoId,
+  getSelectedTodoId,
+} from '../../store';
+
+import './TodoList.scss';
+import { getTodos, getUserById } from '../api/api';
+
+export const TodoList: React.FC = () => {
+  const [inputedValue, setInputedValue] = useState('');
+  const [selectedTodosValue, setSelectedTodosValue] = useState('all');
```
Comment (Denchepornyuk) [inline src/components/TodoList/TodoList.tsx:19]: would be great to create enum for selectedTodosValue

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      case 'active':
+        return todo.completed === false;
+      default:
+        return !todoForRemoving.includes(todo.id);
+    }
+  });
+
+  let filteredTodos = [];
+
+  filteredTodos = [...selectedTodos].filter((todo) => {
```
Comment (Denchepornyuk) [inline src/components/TodoList/TodoList.tsx:72]: the filter is an immutable method so you don't need to make an array copy

File: src/store/index.ts [hunk 5]
```diff
@@ -2,43 +2,81 @@ import { createStore, AnyAction } from 'redux';
...
+};
+
+export type User = {
+  id: number,
+  name: string,
+  email: string,
+  phone: string,
+};
+
 export type RootState = {
```
Comment (Denchepornyuk) [inline src/store/index.ts:59]: move types to separate file

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**